packages/scikits-odes/src/scikits_odes/ode.py

```diff
@@ -306,6 +306,12 @@ def get_info(self):
         else:
             return {}
 
+    def print_stats(self):
+        if hasattr(self._integrator, "print_stats"):
+            self._integrator.print_stats()
+        else:
+            print(f"Method `print_stats` is not implemented for integrator {self._integrator}")
```

Comment:
This should probably throw an exception or use `logging.error`, rather than printing.

Reply:
AFAIK, I have modeled this method based on the method directly above it:
```python
    def get_info(self):
        """
        Return additional information about the state of the integrator.

        Returns
        -------

        A dictionary filled with internal data as exposed by the integrator.
        See the `get_info` method of your chosen integrator for details.

        """
        if hasattr(self._integrator, 'get_info'):
            return self._integrator.get_info()
        else:
            return {}
```

Will this work?
```python
    def print_stats(self):
        if hasattr(self._integrator, "print_stats"):
            self._integrator.print_stats()
        else:
            raise AttributeError(
                f"Method `print_stats` is not implemented for integrator "
                f"'{self._integrator}'")
```

`claude.ai` says that "[this] approach is idiomatic and well-suited to the adapter pattern you're implementing" :-D